lib/src/util.dart

```diff
@@ -16,6 +22,30 @@ dynamic _getSubpath(Map<String, dynamic> node, List<String> pathSegments) {
   }
 }
 
+// build paths which contain collectionId
+@visibleForTesting
+Map<String, dynamic> buildSubpathForCollectionGroup(Map<String, dynamic> root,
+    Map<String, dynamic> node, String collectionId, Map<String, dynamic> result,
+    [String path = '']) {
+  final pathSegments = path.split('/');
+  for (final entry in node.entries) {
+    if (pathSegments.length > 1 &&
+        pathSegments[pathSegments.length - 2] == collectionId) {
+      result[pathSegments.first] = root[pathSegments.first];
```

Comment:
So if the database is like so:

```
{
 someCollection: {
  bar: {
    doc: someData,
  },
  notBar: {
    someOtherDoc: someOtherData,
  },
 }
}
```

It will return the same:

```
{
 someCollection: {
  bar: {
    doc: someData,
  },
  notBar: {
    someOtherDoc: someOtherData,
  },
 }
}
```

Is that ok?